src/lib/FreeC/Pass/DefineDeclPass.hs

```diff
@@ -113,8 +115,8 @@ defineFuncDecl funcDecl = do
     { entrySrcSpan       = IR.funcDeclSrcSpan funcDecl
     , entryArity         = length (IR.funcDeclArgs funcDecl)
     , entryTypeArgs = map IR.typeVarDeclIdent (IR.funcDeclTypeArgs funcDecl)
-    , entryArgTypes      = map IR.varPatType (IR.funcDeclArgs funcDecl)
-    , entryReturnType    = IR.funcDeclReturnType funcDecl
+    , entryArgTypes = map (fromJust . IR.varPatType) (IR.funcDeclArgs funcDecl)
```

Comment:
I'm assuming that the `fromJust` ist needed due to the `Map.lookup` operation yielding `Maybe`. Would it make sense to switch to a total map instead?

Reply:
I don't see a `Map.lookup` here. A [`FuncDecl`](https://freeproving.github.io/free-compiler/docs/master/FreeC-IR-Syntax-FuncDecl.html#t:FuncDecl) contains a list of [`VarPat`](https://freeproving.github.io/free-compiler/docs/master/FreeC-IR-Syntax-Expr.html#t:VarPat)s, in which a [`Type`](https://freeproving.github.io/free-compiler/docs/master/FreeC-IR-Syntax-Type.html#t:Type) is contained in a `Maybe`. `IR.varPatType` and `IR.funcDeclArgs` are just the record field names.
Or do you mean something else?

Reply:
Sorry, my comment was referencing another change but somehow ended up here. I have moved the comment there.